### PR TITLE
78 calc edge cases

### PR DESCRIPTION
**Pull request by kanenathan213**
There's a lot in here...kinda bundled a few different tasks in here. Here are the major themes to watch out for:

- Handling more edge cases in the converter and graph, preventing the UI from breaking and instead, showing an error message
- Styling tweaks to improve responsiveness across device sizes
- Build process improvements: grunt handling watch/compile for Sass (instead of Compass). So now, you can just run 'grunt' in terminal for watch to start. ALSO, I put the compile css into .gitignore. I am going to try to compile it on the server, so that it's not tracked in the git repo (which it doesn't need to be).

Note: I switched over to Atom as my text editor. It automatically removes trailing whitespace, which I think it fine, but it just looks like a bunch more changes in the diff. Just ignore those.

### CODE REVIEW

File: javascript/app.js [hunk 1]
```diff
@@ -500,14 +521,15 @@ FinancialFreedom.controller('TimeToRetirementController', function($scope, Retir
         $scope.retirement.already_retired_shown = retirement_data.can_retire_immediately;
         CreateRetirementGraphService.createRetirementGraph(retirement_data);
     };
-    
+
     $(window).resize(function() {
-        CreateRetirementGraphService.createRetirementGraph(retirement_data);
+        var retirement_data = RetirementCalculatorService.calculateRetirementInfo();
+        $scope.refreshOutput(retirement_data);
```
Comment (bobo333) [inline javascript/app.js]: A useful improvement at some point in the future would be to just redraw the graph instead of re-calculating everything on a window resize. The data shouldn't change, just the drawing of the graph. I think I initially put this in here because I was being lazy and it was easier at the time to re-do everything than split up the drawing and the calculation logic.

File: javascript/create_retirement_graph_service.js [hunk 2]
```diff
@@ -2,6 +2,20 @@ FinancialFreedom.service('CreateRetirementGraphService', ['DateService', functio
 
     this.createRetirementGraph = function(retirement_data) { //graph_points, intersection_point
 
+        if (!retirement_data.can_retire) {
+            d3.select("svg").attr("style","opacity:0.2");
+            d3.select(".tooltip").remove();
+            d3.select(".cannot-retire-message").remove();
```
Comment (bobo333) [inline javascript/create_retirement_graph_service.js]: why does this get removed and re-added in the next step?

File: javascript/create_retirement_graph_service.js [hunk 3]
```diff
@@ -13,24 +27,24 @@ FinancialFreedom.service('CreateRetirementGraphService', ['DateService', functio
         pixels_per_axis_label = 75;
 
         if ($(".tooltip").length > 0) {
-            $( ".tooltip" ).remove();
+            $(".tooltip" ).remove();
```
Comment (bobo333) [inline javascript/create_retirement_graph_service.js]: the trailing space could be removed too, so `$(".tooltip").remove();`

File: javascript/create_retirement_graph_service.js [hunk 4]
```diff
@@ -166,15 +180,23 @@ FinancialFreedom.service('CreateRetirementGraphService', ['DateService', functio
...
 
-        return "You will be able to safely <br>live off passive income in <br> <span class='bold'>" + date.getFullYear() + "</span>, when <br>you have total assets <br>of <span class='bold'>$" + numberWithCommas(asset_need) + "</span>.";
-        
+        if (retirement_data.intersection_point) {
+            var date = retirement_data.intersection_point.x;
+            var expenses = retirement_data.intersection_point.y;
+            var asset_need = 25 * expenses * 12;
+            asset_need = Math.round(asset_need);
+
+            return "You will be able to safely <br>live off passive income in <br> <span class='bold'>" + date.getFullYear() + "</span>, when <br>you have total assets <br>of <span class='bold'>$" + numberWithCommas(asset_need) + "</span>.";
```
Comment (bobo333) [inline javascript/create_retirement_graph_service.js]: generally try to make lines less than 120 characters (some people say 80, but with bigger screens I usually do 12) long so it's easier to read, not a huge deal just something to look for

Comment (bobo333) [inline javascript/create_retirement_graph_service.js]: note, this usually doesn't apply to html (cause that can be more annoying than helpful), I mean just the js files

**Comment by bobo333** [comment]
Just a few comments, for the most part looks good. If some of the edge cases have been addressed, should there be some unit tests added to make sure those edge cases are covered?

File: javascript/create_retirement_graph_service.js (hunk 2 above)
Comment (kanenathan213) [inline javascript/create_retirement_graph_service.js]: It clears the message if one existed prior, so multiple don't get stacked up.

File: javascript/app.js (hunk 1 above)
Comment (kanenathan213) [inline javascript/app.js]: Nice, I added it to Trello
